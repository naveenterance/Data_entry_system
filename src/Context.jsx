import React, { createContext, useContext, useReducer } from "react";

const Context = createContext();

const ContextProvider = ({ children }) => {
  const initialState = {
    name: "",
    email: "",
    submittedData: [],
  };

  const reducer = (state, action) => {
    switch (action.type) {
      case "UPDATE_FIELD":
        return { ...state, [action.field]: action.value };
      case "RESET":
        return { ...state, name: "", email: "" };
      case "ADD":
        return {
          ...state,
          submittedData: [...state.submittedData, action.formData],
        };
      default:
        return state;
    }
  };

  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <Context.Provider value={{ state, dispatch }}>{children}</Context.Provider>
  );
};

export const Contexthook = () => {
  return useContext(Context);
};

export default ContextProvider;
