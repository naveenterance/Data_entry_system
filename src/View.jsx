import React, { useState, useEffect } from "react";

const View = (props) => {
  return (
    <>
      <h2>Submitted Data:</h2>
      <ul>
        {props.entries.map((data, index) => (
          <li key={index}>{`Name: ${data.name}, Email: ${data.email}`}</li>
        ))}
      </ul>
    </>
  );
};

export default View;
