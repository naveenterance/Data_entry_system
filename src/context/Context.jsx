import React, { createContext, useContext, useReducer } from "react";

export const Context = createContext();

export const ContextProvider = ({ children }) => {
  const initialState = {
    firstName: "",
    lastName: "",
    email: "",
    phone: "",
    country: "",
    age: "",
    street: "",
    town: "",
    postcode: "",
    visits: "0",
    entries: [],
  };

  const reducer = (state, action) => {
    switch (action.type) {
      case "UPDATE_FIELD":
        return { ...state, [action.field]: action.value };
      case "RESET":
        return {
          ...state,
          firstName: "",
          lastName: "",
          email: "",
          phone: "",
          country: "",
          age: "",
          street: "",
          town: "",
          postcode: "",
          visits: "0",
        };
      case "ADD":
        return {
          ...state,
          entries: [...state.entries, action.formData],
        };
      default:
        return state;
    }
  };

  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <Context.Provider value={{ state, dispatch }}>{children}</Context.Provider>
  );
};

export const Contexthook = () => {
  return useContext(Context);
};
