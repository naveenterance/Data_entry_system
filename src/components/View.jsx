import React from "react";
import { Contexthook } from "../context/Context";

const View = () => {
  const { state } = Contexthook();

  return (
    <div>
      <h2>View Component</h2>
      <ul>
        {state.entries.map((data, index) => (
          <li key={index}>
            Name: {data.firstName} {data.lastName}, Email: {data.email}, Phone:{" "}
            {data.phone},age: {data.age} ,country: {data.country}, address :{" "}
            {data.street}
            {data.town}
            {data.postcode}
            Visits: {data.visits}
          </li>
        ))}
      </ul>
    </div>
  );
};

export default View;
