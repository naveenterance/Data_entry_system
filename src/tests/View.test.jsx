import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Contexthook, ContextProvider } from "../context/Context";
import View from "../components/View";

jest.mock("../context/Context", () => ({
  Contexthook: jest.fn(),
  ContextProvider: ({ children }) => <div>{children}</div>,
}));

describe("View component", () => {
  it("renders submitted data correctly ", () => {
    const mockentries = [
      {
        firstName: "John",
        lastName: "Doe",
        email: "john@example.com",
        phone: "123456789",
        age: 30,
        country: "USA",
      },
      {
        firstName: "Alice",
        lastName: "Smith",
        email: "alice@example.com",
        phone: "987654321",
        age: 25,
        country: "Canada",
      },
    ];

    Contexthook.mockReturnValue({
      state: { entries: mockentries },
    });

    const { container } = render(
      <ContextProvider>
        <View />
      </ContextProvider>
    );

    mockentries.forEach((data) =>
      ["firstName", "lastName", "email", "phone", "age", "country"].forEach(
        (field) => expect(container).toHaveTextContent(data[field].toString())
      )
    );

    expect(container).toMatchSnapshot();
  });

  it("renders without crashing when no data is submitted  ", () => {
    Contexthook.mockReturnValue({
      state: { entries: [] },
    });

    const { container } = render(
      <ContextProvider>
        <View />
      </ContextProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
