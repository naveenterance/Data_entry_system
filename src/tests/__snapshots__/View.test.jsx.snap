// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View component renders submitted data correctly  1`] = `
<div>
  <div>
    <div>
      <h2>
        View Component
      </h2>
      <ul>
        <li>
          Name: 
          John
           
          Doe
          , Email: 
          john@example.com
          , Phone:
           
          123456789
          ,age: 
          30
           ,country: 
          USA
          , address :
           
          Visits: 
        </li>
        <li>
          Name: 
          Alice
           
          Smith
          , Email: 
          alice@example.com
          , Phone:
           
          987654321
          ,age: 
          25
           ,country: 
          Canada
          , address :
           
          Visits: 
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`View component renders without crashing when no data is submitted   1`] = `
<div>
  <div>
    <div>
      <h2>
        View Component
      </h2>
      <ul />
    </div>
  </div>
</div>
`;
